encoding/encodingtest/series_iterator_split_into_blocks_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package encodingtest
+
+import (
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/encoding"
+	"github.com/m3db/m3db/encoding/m3tsz"
+	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3db/x/xio"
+	"github.com/m3db/m3x/checked"
+	"github.com/m3db/m3x/ident"
+	xtime "github.com/m3db/m3x/time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type Series struct {
+	ID     ident.ID
+	Blocks []SeriesBlock
+}
+
+type SeriesBlock struct {
+	Start          time.Time
+	End            time.Time
+	Replicas       []encoding.MultiReaderIterator
+	ValuesIterator encoding.SeriesIterator
+}
+
+func TestDeconstructAndReconstruct(t *testing.T) {
```

Comment:
suggestion(take/leave): might be more easier to follow if this was written as an `Example...`. Feel free to ignore if you don't think it's worth the hassle.

Reply:
Sounds good, didn't know this was a thing!